### ISSUE TITLE
Both xpip and pip fail to run

### ISSUE THREAD

**Issue by seanfarley** [xonssh/xxh#22]
```
>>> cd ~/.xxh                                                                                                                                                         
>>> xpip install xonsh-vox-tabcomplete                                                                                                                                             
usage: -c [-h] [-V] [-c COMMAND] [-i] [-l] [--config-path CONFIG_PATH] [--rc RC [RC ...]] [--no-rc] [--no-script-cache] [--cache-everything] [-D ITEM]
          [--shell-type {b,best,d,dumb,ptk,ptk1,ptk2,prompt-toolkit,prompt_toolkit,prompt-toolkit1,prompt-toolkit2,rand,random,rl,readline}] [--timings]
          [script-file] ...
-c: error: unrecognized arguments: -m
                                                                                                                                                    
>>> pip install xonsh-vox-tabcomplete                                                                                                                                              
Traceback (most recent call last):
  File "/tmp/.mount_xonshS434io/usr/python/lib/python3.8/site-packages/xonsh/main.py", line 399, in main
    args = premain(argv)
  File "/tmp/.mount_xonshS434io/usr/python/lib/python3.8/site-packages/xonsh/main.py", line 352, in premain
    env = start_services(shell_kwargs, args)
  File "/tmp/.mount_xonshS434io/usr/python/lib/python3.8/site-packages/xonsh/main.py", line 283, in start_services
    execer = Execer(
  File "/tmp/.mount_xonshS434io/usr/python/lib/python3.8/site-packages/xonsh/execer.py", line 63, in __init__
    load_builtins(execer=self, ctx=xonsh_ctx)
  File "/tmp/.mount_xonshS434io/usr/python/lib/python3.8/site-packages/xonsh/built_ins.py", line 1288, in load_builtins
    builtins.__xonsh__.load(execer=execer, ctx=ctx)
  File "/tmp/.mount_xonshS434io/usr/python/lib/python3.8/site-packages/xonsh/built_ins.py", line 1360, in load
    self.env = Env(default_env())
  File "/tmp/.mount_xonshS434io/usr/python/lib/python3.8/site-packages/xonsh/environ.py", line 1402, in __init__
    self[key] = val
  File "/tmp/.mount_xonshS434io/usr/python/lib/python3.8/site-packages/xonsh/environ.py", line 1571, in __setitem__
    self.replace_env()
  File "/tmp/.mount_xonshS434io/usr/python/lib/python3.8/site-packages/xonsh/environ.py", line 1439, in replace_env
    os_environ.update(self.detype())
  File "/tmp/.mount_xonshS434io/usr/python/lib/python3.8/site-packages/xonsh/environ.py", line 1424, in detype
    deval = ensurer.detype(val)
  File "/tmp/.mount_xonshS434io/usr/python/lib/python3.8/site-packages/xonsh/tools.py", line 1103, in detype
    return x.detype()
  File "/tmp/.mount_xonshS434io/usr/python/lib/python3.8/site-packages/xonsh/environ.py", line 364, in detype
    style = self.style
  File "/tmp/.mount_xonshS434io/usr/python/lib/python3.8/site-packages/xonsh/environ.py", line 396, in style
    style_name = self.style_name
  File "/tmp/.mount_xonshS434io/usr/python/lib/python3.8/site-packages/xonsh/environ.py", line 386, in style_name
    env = builtins.__xonsh__.env
AttributeError: 'XonshSession' object has no attribute 'env'
Xonsh encountered an issue during launch
Failback to /bin/sh
$ 
```

Any idea what I'm doing wrong?

**Comment by anki-code** [comment]
Hey @seanfarley, thank you for the reporting!

0. You're using legacy call for pip. The recommended is:
```
python3 -m pip install --upgrade pandas
```
Probably we should add an alias for it.

1. I've fixed xpip and release xxh 0.2.11. Please update your xxh:
```
python3 -m pip install --upgrade xxh-xxh
```
Then update host:
```
$ xxh yourhost +iv
yourhost$ xpip --version                                                                                                                                                                                                                       
pip 19.3.1
```
You can update pip easily: `xpip install --upgrade pip`

2. To install xontrib just run:
```
xpip install --upgrade xonsh-vox-tabcomplete
```
After that `xontrib list` could shows that the xontrib is not installed but `pip list` shows that it is. I think this is because `xontrib` is not update the modules before list the xontribs. There are two ways to workaround: 
* Manually run `import importlib; importlib.reload(sys.modules['xontrib'])`
* OR reconnect xxh

After that the xontrib will have "installed" status in `xontrib list`.

3. Let's load:
```
$ xontrib load vox_tabcomplete
xonsh: For full traceback set: $XONSH_SHOW_TRACEBACK = True
NameError: name '__xonsh_completers__' is not defined
```
This error for this xontrib I've got also on my local but I'm not the user of this xontrib and I suppose you know what to do with xonsh_completers, isn't it? :)

**Comment by anki-code** [comment]
I've fixed `pip` also in xxh 0.2.12. Please update your xxh:
```
python3 -m pip install --upgrade xxh-xxh
```
Looks like closed.

**Comment by seanfarley** [comment]
Thanks for the quick response! Yes, I can now `xpip` install :-)

**Comment by anki-code** [comment]
```
AttributeError: 'XonshSession' object has no attribute 'env'
Xonsh encountered an issue during launch
Failback to /bin/sh
```
@seanfarley I've found the case of this error and it was fixed in xxh 0.3.0

**Comment by anki-code** [comment]
```
AttributeError: 'XonshSession' object has no attribute 'env'
Xonsh encountered an issue during launch
Failback to /bin/sh
```
@seanfarley thank you for reporting this! It's related with https://github.com/xonsh/xonsh/issues/3055 and I'm going to fix it in xxh 0.3.1
